vllm/worker/model_runner.py

```diff
@@ -266,20 +266,23 @@ def _prepare_prompt(
                 # Prefix is not supported with sliding_window
                 context_len = len(computed_block_nums) * self.block_size
                 prompt_tokens = prompt_tokens[context_len:]
-                prefix_block_tables.append(computed_block_nums)
+                if self.attn_backend.get_name() == "flash-attn":
+                    block_table = seq_group_metadata.block_tables[seq_id]
+                else:
+                    block_table = computed_block_nums
```

Comment:
this suggests to me we should use some abstraction here instead of if...else branching

Reply:
Good point. Agreed. For now, I added a note on why we need this `if`, and added TODO asking for a better abstraction.

Reply:
I think the simplest solution here is to fix the Triton kernel so that it aligns with the other backends' APIs. I will leave this for future PR though.